### ISSUE TITLE
OneSignal not compatible with firebase messaging 

### ISSUE THREAD

**Issue by MohammedAlimoor** [OneSignal/OneSignal-Flutter-SDK#419]
**Description:**

when i'am add  oneSignal-Flutter-SDK to the project the the  firebase messaging  handler not working  `FirebaseMessaging.onMessage` or  `FirebaseMessaging.onMessageOpenedApp` on ios 

**Environment**

> flutter 2.0.6

> pubspec.yaml

```
  firebase_core: "^1.1.0"
  firebase_analytics: any
  firebase_messaging: 10.0.2
  firebase_crashlytics: any
  onesignal_flutter: 3.0.0
```

> AppDelegate.swift

```
@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate , OPPCheckoutProviderDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
        GeneratedPluginRegistrant.register(with: self)

    var flutter_native_splash = 1
    UIApplication.shared.isStatusBarHidden = false

    if #available(iOS 10.0, *) {
        UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
      }
         return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
  }
```

**Steps to Reproduce Issue:**

just send a push notification from firebase

**Comment by emawby** [comment]
Hello can you try the workaround posted in this issue? https://github.com/OneSignal/OneSignal-Flutter-SDK/issues/348

**Comment by emawby** [comment]
@MohammedAlimoor are you still experiencing issues with this?

**Comment by MohammedAlimoor** [comment]
yes

**Comment by emawby** [comment]
@MohammedAlimoor Have you tried setting your AppDelegate as a `UNUserNotificationCenterDelegate`?

**Comment by jkasten2** [comment]
@MohammedAlimoor Google has shipped the fix for this in [firebase_messaging 11.4.2](https://pub.dev/packages/firebase_messaging/changelog#1142).
The following change log entry relates to this:
* FIX: Swizzle check for FlutterAppLifeCycleProvider instead of UNUserNotificationCenterDelegate (#8822). ([81f6b274](https://github.com/firebase/flutterfire/commit/81f6b2743b99e47c16fc3ee13cc1e7e6e7982730))
